docs/csharp/programming-guide/classes-and-structs/local-functions.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Local functions (C# Programming Guide) | Microsoft Docs"
+ms.date: "2017-06-14"
+ms.prod: .net
+ms.technology: 
+  - "devlang-csharp"
+ms.topic: "article"
+dev_langs: 
+  - "CSharp"
+helpviewer_keywords: 
+  - "fields [C#]"
+author: "rpetrusha"
+ms.author: "ronpet"
+---
+# Local functions (C# Programming Guide)
+
+Starting with C# 7, C# supports *local functions*. Local functions are private methods of a type that are nested in another member. They can only be called from their containing member. Local functions can be declared in and called from:
+
+- Methods, especially iterator methods and async methods
+- Constructors
+- Property accessors
+- Event accessors
+- Anonymous methods
+- Lambda expressions
+- Finalizers
+- Other local functions
+
+However, local functions can't be declared inside an expression-bodied member.
+
+> [!NOTE]
+> In some cases, you can use a lambda expression to implement functionality also supported by a lambda expression. For a comparison, see [Local functions compared to Lambda expressions](../../local-functions-vs-lambdas.md).
+
+Local functions make the intent of your code clear. Anyone reading you code can see that the method is private and is not callable except by the containing method. For team projects, they also make it impossible for another developer to mistakenly call the method directly from elsewhere in the class or stuct.
+ 
+## Local function syntax
+
+A local function is defined as a nested method inside a containing member. Its definition has the following syntax:
+
+```csharp
+<return-value> <method-name> <parameter-list>
+```
```

Comment:
I don't think C# syntax highlighting is appropriate for this code block, especially since it's likely going to highlight `return` as a keyword (it does that on GiHub).